v2/errors.go

```diff
@@ -53,9 +53,12 @@ func (e HTTPStatusCodeError) Error() string {
 // IsHTTPError returns whether the error represents an HTTPStatusCodeError.  A
 // client method returning an HTTP error indicates that the broker returned an
 // error code and a correctly formed response body.
-func IsHTTPError(err error) bool {
-	_, ok := err.(HTTPStatusCodeError)
-	return ok
+func IsHTTPError(err error) (*HTTPStatusCodeError, bool) {
+	statusCodeError, ok := err.(HTTPStatusCodeError)
```

Comment:
You also need to test conversion to a pointer - if you are passed a pointer to HTTPStatusCode, this will fail.  See: https://play.golang.org/p/3rgurzoIaA